### ISSUE TITLE
Lua Editor: Unable to connect to the O3DE Editor

### ISSUE THREAD

**Issue by LB-TomaszSerwanski** [o3de/o3de#9160]
**Describe the bug**
The Lua Editor cannot be connected to the O3DE Editor. On the first launch of the Lua Editor, the only available options in Target list are AssetProcessor and AssetBuilder. When the Lua Editor is restarted, these are no longer available and the Target list is empty.
Additionally, when the Lua Editor is closed for the first time, the following Assert appears in the Console:

```
Trace::Assert
 D:\Git\o3de\Code\Framework\AzCore\AzCore/std/parallel/thread.h(99): (12188) 'class AZStd::thread &__cdecl AZStd::thread::operator =(class AZStd::thread &&)'
You must call detach or join before you delete/move over the current thread!
```

Please refer to the attached video and assert.txt file for more details.

**Steps to reproduce**
1. Launch O3DE Editor.
2. Launch Lua Editor (Tools -> Lua Editor).
3. Switch focus to the O3DE Editor (due to #7237).
4. Switch focus back to the Lua Editor.
5. Click the Target button to attempt to connect to the O3DE Editor.
6. Restart the LUA editor.
7. Repeat steps 3-6.

**Expected behavior**
The O3DE Editor appears on the Target list and can be connected to.

**Actual behavior**
The O3DE Editor does not appear on the Target list and cannot be connected to.

**Screenshots/Video**

https://user-images.githubusercontent.com/86953108/165092336-29204b7d-1317-45eb-a8c4-d05e32263236.mp4

**Found in Branch**
Development (f590593)

**Desktop/Device:**
 - Device: PC
 - OS: Windows
 - Version 10
 - CPU AMD Ryzen 5 3600
 - GPU Nvidia RTX 2060 Super
 - Memory 16GB

**Attachments**
[Assert.txt](https://github.com/o3de/o3de/files/8554668/Assert.txt)

**Comment by lmbr-pip** [comment]
@AMZN-puvvadar @robhaco Any ideas on this one? 

```
 D:\Git\o3de\Code\Framework\AzCore\AzCore/std/parallel/thread.h(99): (12188) 'class AZStd::thread &__cdecl AZStd::thread::operator =(class AZStd::thread &&)'
You must call detach or join before you delete/move over the current thread!
```

**Comment by AMZN-puvvadar** [comment]
> @AMZN-puvvadar @robhaco Any ideas on this one?
> 
> ```
>  D:\Git\o3de\Code\Framework\AzCore\AzCore/std/parallel/thread.h(99): (12188) 'class AZStd::thread &__cdecl AZStd::thread::operator =(class AZStd::thread &&)'
> You must call detach or join before you delete/move over the current thread!
> ```

I re-exmained code to sanity check. TargetManagement currently uses two threads. One is the thread used to poll to connect to the LuaIDE. That reliably joins on destruction which should be triggered from the component's Deactivate. The other is the outbox processing thread with also joins on deactivate.

If Deactivate was not being called before destruction, the destructor would assert. Another possibility is if there were somehow more than one TargetManagementComponent.

**Comment by robhaco** [comment]
Updated note, wasn't able to repro the connection issue, but was able to get the assert to occur.  This didn't seem to stop any functionality, though, and the LuaIDE was still able to connect.  It's likely these are two separate issues.

**Comment by LB-TomaszSerwanski** [comment]
We have verified that the connection problem only occurs on a prebuilt-sdk engine. On the source engine the Lua Editor can be successfully connected to the O3DE Editor (the assert still occurs on both). 

In this case, should the assert be reported as a separate issue?

**Comment by AMZN-puvvadar** [comment]
To update, I've now also seen the assert but it's sporadic at best and I'm not yet sure which application it's occurring in. It's likely a race condition.

**Comment by LB-TomaszSerwanski** [comment]
As requested, the assert has been reported in a separate issue (#9612).

**Comment by AMZN-puvvadar** [comment]
A fix has been proposed for the assert in the separate issue mentioned above.

**Comment by AMZN-puvvadar** [comment]
#9612 has been closed. @LB-TomaszSerwanski is there an outstanding issue remaining given that fix and that connecting seems to work on development otherwise?

**Comment by robhaco** [comment]
@LB-TomaszSerwanski when possible, can you check that this issue is still occurring.  When testing the fix related to #9612 with the development installer, I found I was able to connect the LuaIDE to the Editor without issue.  Just want to confirm you're seeing the same.  Thanks!

**Comment by LB-TomaszSerwanski** [comment]
@robhaco, @AMZN-puvvadar, we have verified that it is not possible to connect LUA Editor to O3DE Editor on Linux (both source and prebuilt-sdk engine). The target list only shows "Disconnect". This was confirmed on the latest Development (146b2c8). Please note that connecting to the Editor works as expected on Windows.

**Comment by robhaco** [comment]
@LB-TomaszSerwanski I believe this is now fixed with the changes related to the remote tools connection gem.  I have been able to connect without issue since.  Note: The remote tools connection gem is now required for a project to be able to connect in general.

**Comment by LB-ArturZieba** [comment]
Issue verified as fixed on the Development (23cc830) build on both the Windows and Linux.
